### DISCUSSION TITLE
HttpContext is empty in onRecordsListRequest hook.

### DISCUSSION THREAD

**Discussion by drhuster** [pocketbase/pocketbase#4391] [Q&A] [answered]
My use case is I am looking to block requests coming from urls other than a whitelist of urls that I have determined. Looking at the documentation, I think I could potentially use an event hook. Following the documentation here: https://pocketbase.io/docs/js-event-hooks/#onrecordslistrequest I have added a custom hook inside the `pb_hooks` folder which is mostly a copy of what is in the documentation

```
onRecordsListRequest((e) => {
  console.log("RECORD LIST");
  console.log("CONTEXT", JSON.stringify(e.httpContext));
  console.log("RESULT", JSON.stringify(e.result));
});
```

I get data back for the result, however, for the httpContext, it is coming back as an empty object `{}`.

I am doing this all locally. So I have a simple vue.js app served locally that uses the pocketbase client sdk to request data from my collections from pocketbase running locally as well. 

Am I missing something? Or am I misusing how the hook should be used?

I did search for similar questions and came up to this discussion (https://github.com/pocketbase/pocketbase/discussions/4077) but the answer made it seem like I should be seeing the httpContext.

**Comment by ganigeorgiev** [comment, marked as answer]
`httpContext` is not a plain JavaScript object and the empty string representation means that it doesn't have any public serializable fields.

For the available `httpContext` methods, please refer to https://pocketbase.io/jsvm/interfaces/echo.Context.html.

**Reply by ganigeorgiev** [reply]
_You can also place at the top of your hooks file `/// <reference path="../pb_data/types.d.ts" />`.
This will instruct your editor language server to assist you with code completion and documentation._

**Reply by ganigeorgiev** [reply]
Also about:

> I am looking to block requests coming from urls other than a whitelist of urls

I'm not sure that I understand what do you mean by "coming from urls" but if you mean the request Origin header you can't rely on it.

You can enable CORS if you want to block browser requests coming from other hosts by specifying the `--origins` flag when starting the executable (eg. `./pocketbase serve --origins=https://example1.com,https://example2.com`; use `--help` for list with all options) but note that this will not prevent others to send requests to your api using curl, Postman, Insomnia, etc.

_edit: updated the origins example to use proper urls_

**Reply by drhuster** [reply]
This answers my original ask. Thanks so much! I understand what you mean about the curl, Postman, etc. I am guessing the default way to protect data is through the API rules?

**Reply by ganigeorgiev** [reply]
> I am guessing the default way to protect data is through the API rules?

Yes.
